titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiFacade.java

```diff
@@ -18,26 +18,49 @@
 
 import com.netflix.titus.runtime.connector.kubernetes.v1.V1OpportunisticResource;
 import io.kubernetes.client.informer.SharedIndexInformer;
-import io.kubernetes.client.openapi.ApiClient;
-import io.kubernetes.client.openapi.apis.CoreV1Api;
-import io.kubernetes.client.openapi.apis.CustomObjectsApi;
+import io.kubernetes.client.openapi.ApiException;
 import io.kubernetes.client.openapi.models.V1Node;
 import io.kubernetes.client.openapi.models.V1PersistentVolume;
 import io.kubernetes.client.openapi.models.V1PersistentVolumeClaim;
 import io.kubernetes.client.openapi.models.V1Pod;
+import reactor.core.publisher.Mono;
 
+/**
+ * {@link KubeApiFacade} encapsulates Kube Java, except the entity model and the informer API. The latter is
+ * provided as a set of interfaces (unlike ApiClient or CoreV1Api), so it is easy to mock in the test code.
+ */
 public interface KubeApiFacade {
 
-    ApiClient getApiClient();
-
-    CoreV1Api getCoreV1Api();
+    // Nodes
 
-    CustomObjectsApi getCustomObjectsApi();
+    void deleteNode(String nodeName) throws ApiException;
```

Comment:
Maybe we shouldn't leak the underlying client's ApiException? Thoughts?

Reply:
Good idea. I will create a wrapper